File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.java
import java.util.*;


class Solution {
    static char[] alphabets  = {'A', 'E', 'I', 'O', 'U'};
    public int solution(String word) {
        
    List<String> list = new ArrayList<>();
    for(int i=0; i<alphabets.length; i++){
      dfs(list, String.valueOf(alphabets[i]));
    }
        return list.indexOf(word)+1;
    }
    void dfs(List<String> list, String word){
    if(word.length() > 5) return;
    
    list.add(word);

    for(int i=0; i<alphabets.length; i++){
      dfs(list, word + alphabets[i]);  
    }
    
  }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    let answer = 0;
    let cnt = 0;
    
    const order = ["A", "E", "I", "O", "U"];
    
    const dfs = (str) => {
        if (str === word) return true;
        if (str.length >= 5) return false;
        
        for (const char of order) {
            cnt++;
            if (dfs(str + char)) {
                answer = cnt;
                return true;
            }
        }
        return false;
    }
    
    dfs("")
    
    return answer;
}